### ISSUE TITLE
Optional argument incorrectly filled

### ISSUE THREAD

**Issue by hillin** [dotnet/command-line-api#2050]
```csharp
var moveCommand = new Command("move");

var xPositionArgument = new Argument<int>("x")
{
    Description = "X position."
};
moveCommand.AddArgument(xPositionArgument);

var yPositionArgument = new Argument<int>("y")
{
    Description = "Y position."
};
moveCommand.AddArgument(yPositionArgument);

var zPositionArgument = new Argument<int?>("z")
{
    Description = "Z position."
};
moveCommand.AddArgument(zPositionArgument);

moveCommand.SetHandler(
    (x, y, z) =>
    {
        Console.WriteLine($"moving to {x}, {y}, {z}");
    },
    xPositionArgument,
    yPositionArgument,
    zPositionArgument);
```

`y` value will be used for `z` if `z` is omitted:

```
> move 300 300 400
moving to 300, 300, 400
> move 300 300
moving to 300, 300, 300
```

**Comment by elgonzo** [comment]
This has already been fixed. However, the fix came _after_ the release of 2.0.0-beta4.22272.1, which is still the newest beta available on the official nuget.org feed but is already about 8 month old by now.

The current daily build 2.0.0-beta4.23113.1 does not exhibit this issue anymore. Note that as a daily build, it is not available through the nuget.org feed but through the daily builds feed mentioned here: https://github.com/dotnet/command-line-api#packages. (Also note that the the API surface of the library still changes, and with newer builds, `moveCommand.AddArgument` would need to be replaced with `moveCommand.Arguments.Add`)

**Comment by hillin** [comment]
Thanks! Guess I'll wait for the next beta then.
